src/common/pluginloader.js

```diff
@@ -79,19 +79,63 @@ function onScriptLoadingComplete (moduleList, finishPluginLoading) {
 function handlePluginsObject (path, moduleList, finishPluginLoading) {
     // Now inject the scripts.
     var scriptCounter = moduleList.length;
+    var modulesWithProblems = [];
 
     if (!scriptCounter) {
         finishPluginLoading();
         return;
     }
+
+    function callOnScriptLoadingComplete () {
+        modulesWithProblems.forEach(function (elem) {
+            var foundIndex = -1;
+            var modulesLength = moduleList.length;
+            var i = 0;
+            // find module with load problems in module list. Use while loop to exit iteration early.
+            while (i < modulesLength && foundIndex === -1) {
+                if (moduleList[i].id === elem) {
+                    foundIndex = i;
+                }
+                i++;
+            }
+            if (foundIndex !== -1) {
+                // delete module with load problem from module list
+                moduleList.splice(foundIndex, 1);
+            }
+        });
+        onScriptLoadingComplete(moduleList, finishPluginLoading);
```

Comment:
```suggestion
        var loadedModules = moduleList.filter(function (m) {
            return modulesWithProblems.indexOf(m.id) === -1;
        });
        onScriptLoadingComplete(loadedModules, finishPluginLoading);
```
That should get the job done as well while making the intention clearer. Plus we do not modify the original array.

Reply:
I can also agree to this version. Reasons why I decided to implement differently is that
the version I implemented is faster as it in the outer loop only iterates over the smaller array of modules with problems. If there are no problems then no code is executed. But this does not matter that much as the module list is most likely just containing <100 elements. 
I'll wait on your comment to that and I am open to accept the suggestion.

Reply:
I don't expect any noticeable slowdown due to the single iteration over all modules. If that should turn out to be a problem, we can still fast-track the no-error case by doing
```js
var loadedModules = modulesWithProblems.length === 0
    ? moduleList
    : moduleList.filter(function (m) {
        return modulesWithProblems.indexOf(m.id) === -1;
    });
```